commit fb426aa31baeb8315f68be8c48987cb77252af74
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-03-11T16:42:14+00:00 (7 months ago)
    
    TOOLS-2181 buildimage target should bail out early for non-zfs components
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/tools/mk/Makefile.targ b/tools/mk/Makefile.targ
index 6744fe8..33a33a4 100644
--- a/tools/mk/Makefile.targ
+++ b/tools/mk/Makefile.targ
@@ -425,8 +425,18 @@ endif
 
 endif
 
+#
+# We add a guard to ensure that users haven't invoked 'make buildimage'
+# by accident. It's better to bail out early with an clear error message
+# if a user has not declared that their component should $ENGBLD_USE_BUILDIMAGE.
+#
 .PHONY: buildimage
 buildimage: stamp-buildimage-prep release $(AGENTS:%=%-prebuilt)
+	if [[ "$(ENGBLD_USE_BUILDIMAGE)" != "true" ]]; then \
+		echo "This component is not configured to use buildimage."; \
+		echo "Set 'ENGBLD_USE_BUILDIMAGE=true' in the component Makefile otherwise."; \
+		exit 1; \
+	fi
 	mkdir -p $(ENGBLD_BITS_DIR)/$(NAME)
 	mkdir -p $(BUILDIMAGE_STAGEDIR)
 	cd $(BUILDIMAGE_STAGEDIR) ; \
